### ISSUE TITLE
TextServer StringGetWordBreaks unable to process special characters

### ISSUE THREAD

**Issue by Squisics** [godotengine/godot#84206]
### Godot version

4.1.1.stable (.net version)

### System information

Windows 10

### Issue description

I'm using a Label3D to have text in 3d space since there doesn't seem any other way to do 2d controls in 3d space compatible with godot versions 4+.

As part of this i can only constrain Label3D by width, not height, thus it wraps text, but doesnt scroll or cut text at the bottom, nor does it have any ability to query the word-breaks used by the control itself to be able to modify text via more cumbersome means to create an effect like text scrolling as more is entered on say a terminal.

Thus I ended up stumbling upon TextServer, and found I can reasonably successfully calculate where the word-breaks will be by supplying "string_get_word_breaks" with the text from the label3d and analysing the array returned.

However I found a weird scenario whereby if non-alphanumerical characters (i havent tested every character, mostly '.', '#', and so on...) are used around where the text would wrap, the TextServer::StringGetWordBreaks command doesn't break those characters.

It's possible there might be other ways to achieve this? But this is the only path I've found exposed to c# for label3D.

### Steps to reproduce

```
int[] line_breaks= TextServerManager.GetPrimaryInterface().StringGetWordBreaks(
    "text\nwith lots of ...\n...\n........................................................................................................................................................\n... that should break at 20 characters"
    ,charsPerLine: 20
);
for(int i= 0; i <line_breaks.Length; i+=2){
    debug.log( $"{line_breaks[i]}->{line_breaks[i+1]}" );
}
```
_(replacing "debug.log" with however you echo text when debugging)_

output:
0->4
5->21
22->25
**26->178**
179->194
195->206
207->217

Notice the 26->178 row, which is unable to break the word when it involves special characters either side of where it would break.
This also seems to be the case if you use alphanumerical characters that fill a row that should word-break, if you end the line with special character like '.' or '#' for example.

### Minimal reproduction project

N/A

**Comment by bruvzg** [comment]
> I'm using a Label3D to have text in 3d space since there doesn't seem any other way to do 2d controls in 3d space compatible with godot versions 4+.

See https://github.com/godotengine/godot-demo-projects/tree/master/viewport/2d_in_3d

`string_get_word_breaks` is not suitable for displaying text, it's used only for the simple things like output to console and text-to-speech word boundaries.

**Comment by Squisics** [comment]
Thanks bruvzg for the info and sanity-check.

Yeah I saw that project, but discussions around it that I saw implied it was for godot 3.5 so I had skipped past it. But I'll try it in 4.1-4.2 anyway to confirm if it still works since yes something like that would be much better than trying to hackishly get label3D into doing this kind of thing.

And OK then I'll only use ```string_get_word_breaks``` for situations where I have control over the text input then.
